libraries/chain/include/scorum/chain/tasks_base.hpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include <vector>
+#include <functional>
+#include <boost/type_index.hpp>
+#include <fc/log/logger.hpp>
+
+namespace scorum {
+namespace chain {
+
+template <typename ContextType> struct task_censor_i
+{
+    virtual ~task_censor_i()
+    {
+    }
+
+    virtual bool is_allowed(ContextType& ctx) = 0;
+    virtual void apply(ContextType& ctx) = 0;
+};
+
+class data_service_factory_i;
+
+template <typename ContextType = data_service_factory_i> class task
```

Comment:
Add censor as a second parameter to template and instantiate it here instead of using censor_type* and set_censor().
Implement default dummy censor.